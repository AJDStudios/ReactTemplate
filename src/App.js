import React from 'react';

function App() {
  const name = 'AD';
  const age = 33;
  const city = 'Gloucester';

  return (
    <div>
      <h1>Welcome to My template</h1>
      <p>Name: {name}</p>
      <p>Age: {age}</p>
      <p>City: {city}</p>
    </div>
  );
}

export default App;
