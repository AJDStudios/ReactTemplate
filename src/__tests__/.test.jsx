import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom/extend-expect';

test('renders the app component', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('does not contain redundant code or comments', () => {
  const { queryByText } = render(<App />);
  expect(queryByText('<!-- Notice the use of %PUBLIC_URL% in the tags above.')).toBeNull();
});

//note that when you remove the splashscreen or update it, this test will also need to be removed or updated
test('renders the splash screen', () => {
  const { getByTestId } = render(<App />);
  const splashScreen = getByTestId('splash-screen');
  expect(splashScreen).toBeInTheDocument();
});

//note that when you remove the splashscreen or update it, this test will also need to be removed or updated
test('displays the correct text in the splash screen', () => {
  const { getByText } = render(<App />);
  const splashText = getByText('An AJDStudios production');
  expect(splashText).toBeInTheDocument();
});

// Add more tests as needed
