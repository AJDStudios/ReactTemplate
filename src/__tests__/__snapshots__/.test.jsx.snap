// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app component 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="splash-screen"
      data-testid="splash-screen"
    >
      <div
        class="splash-header"
      >
        An AJDStudios production
      </div>
    </div>
  </div>
</DocumentFragment>
`;
